src/common/signalk/SKNMEA2000Parser.cpp

```diff
@@ -208,6 +208,38 @@ const SKUpdate& SKNMEA2000Parser::parse127250(const SKSourceInput& input, const
   return _invalidSku;
 }
 
+// *****************************************************************************
+// PGN 127257 Attitude Yaw, Pitch, Roll
+//  - Yaw                   Heading in radians.
+//  - Pitch                 Pitch in radians. Positive, when your bow rises.
+//  - Roll                  Roll in radians. Positive, when tilted right.
+// *****************************************************************************
+const SKUpdate& SKNMEA2000Parser::parse127257(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
+  unsigned char sid;
+  double yaw   = N2kDoubleNA;
+  double pitch = N2kDoubleNA;
+  double roll  = N2kDoubleNA;
+
+  if (ParseN2kPGN127257(msg, sid, yaw, pitch, roll)) {
+    SKUpdateStatic<1> *update = new SKUpdateStatic<1>();
+    update->setTimestamp(timestamp);
+
+    SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
+    update->setSource(source);
+    // TODO: check if it is possible to set NAN as value to SKUpdate
```

Comment:
Setting NAN should work and seems like the best solution to me. We will need to make sure that when we serialize the data to JSON we skip fields that have the value NAN.